README.md

```diff
@@ -24,23 +29,62 @@
 
 Almost all the operation can be supported in form of trees in a convenient way to simplify the structure processing when the calculation is tree-based.
 
-## Installation
+## Outline
+
+* [Overview](#overview)
+* [Getting Started](#getting-started)
+    * [Prerequisite](#prerequisite)
+    * [Installation](#installation)
+    * [Quick Usage](#quick-usage)
+    * [Tutorials](#tutorials)
+* [Speed Performance](#speed-performance)
+* [Change Log](#change-log)
+* [Join and Contribute](#join-and-contribute)
+* [Citation](#citation)
+* [License](#license)
+
+## Overview
+
+When we build a complex nested structure, we need to model it as a tree structure, and the native list and dict in Python are often used to solve this problem. However, it takes a lot of codes and some complex and non-intuitive calculation logics, which is not easy to modify and extend related code and data, and parallelization is more impossible.
+
+Therefore, we need a kind of more proper data container, named `TreeValue`. It is designed for solving the follow problems:
+
```

Comment:
whether to add our overview layer graph

Reply:
I think it is not necessary, and the graph's is verrrrrry long which may break the view of readme.

May be hushuai can given some suggestions?